### PR TITLE
Merge Kousa Dogwood changes

### PR DESCRIPTION
**Pull request by jmcphers**
Merges changes from upstream vscode-server Kousa Dogwood branch into Positron's 2024.11 prerelease branch.

### CODE REVIEW

**Comment by jmcphers** [comment]
@sharon-wang can you verify that this works as intended in Workbench?

File: package.json [hunk 1]
```diff
@@ -255,6 +255,10 @@
     "yaserver": "^0.4.0"
   },
   "resolutions": {
+    "semver": "^7.5.4",
+    "micromatch": "^4.0.8",
+    "braces": "^3.0.3",
+    "ws": "^7.5.10",
```
Comment (sharon-wang) [inline package.json:261]: I checked out this branch and ran `yarn`, which resulted in some `yarn.lock` changes I think we'll want to commit.

File: yarn.lock [hunk 2]
```diff
@@ -2827,23 +2827,7 @@ brace-expansion@^2.0.1:
   dependencies:
     balanced-match "^1.0.0"
 
-braces@^2.3.1, braces@^2.3.2:
```
Comment (sharon-wang) [inline yarn.lock:2830]: I checked out this branch and ran `yarn`, which resulted in some `yarn.lock` changes I think we'll want to commit.

(posted this comment to package.json by accident!)

**Comment by petetronic** [comment]
The history review from upstream is a little involved as a few changes had been cherry picked already, but stepping through the range of history from upstream did confirm this covers the set of changes, less any irrelevant build files that positron does not use.

**Comment by sharon-wang** [comment]
### Workbench Extension v1.5.17 installed :heavy_check_mark: 
image

### Run App: Shiny App :heavy_check_mark: 
image

### Other tests
- opening up data in the data explorer
- python and R plots

**Review by sharon-wang — APPROVED** [review]
LGTM!

**Review by petetronic — APPROVED** [review]

**Pull request merged**